internal/reverseproxy/config/config.go

```diff
@@ -216,120 +178,118 @@ type ProxyUser struct {
 
 // ParseConfig - Parses a given proxy config map
 func (proxyConfig *ProxyConfig) ParseConfig(proxyConfigMap ProxyConfigMap, k8sUtils k8sutils.UtilsInterface) error {
-	proxyMode := proxyConfigMap.Mode
-	proxyConfig.Mode = proxyMode
 	proxyConfig.Port = proxyConfigMap.Port
 	fmt.Printf("ConfigMap: %v\n", proxyConfigMap)
 
-	if proxyMode == StandAlone {
-		config := proxyConfigMap.StandAloneConfig
-		if config == nil {
-			return fmt.Errorf("proxy mode is specified as StandAlone but unable to parse config")
+	config := proxyConfigMap.StandAloneConfig
+	if config == nil {
+		config = proxyConfigMap.Config
+	}
+	if config == nil {
+		return fmt.Errorf("unable to parse config")
+	}
```

Comment:
Is this correct? Seems to be the same test.

Reply:
In order to be backwards compatible, we first check to see if `standaloneConfig` (old field name) is being used. If not, check if `config` (new field name) is set. If this value is still nil, then we fail parsing the configuration.